libs/designsystem/src/lib/directives/floating/floating.directive.ts

```diff
@@ -0,0 +1,262 @@
+
+  public ngOnInit(): void {
+    this.addFloatStylingToHostElement();
+    this.updateHostElementPosition();
+    autoUpdate(
+      this.reference.nativeElement,
+      this.elementRef.nativeElement,
+      this.updateHostElementPosition.bind(this)
+    );
+  }
+
+  /* Should be accessible for programmatically setting display */
+  public show(): void {
+    if (this.isDisabled) {
+      return;
+    }
+
+    this.renderer.setStyle(this.elementRef.nativeElement, 'display', 'block');
+    this.isShown = true;
+  }
+
+  /* Should be accessible for programmatically setting display */
+  public hide(): void {
+    if (this.isDisabled) {
+      return;
+    }
+
+    this.renderer.setStyle(this.elementRef.nativeElement, 'display', 'none');
+    this.isShown = false;
+  }
+
+  /* Should be accessible for programmatically setting display */
+  public toggleShow(): void {
+    this.isShown ? this.hide() : this.show();
+  }
+
+  private addFloatStylingToHostElement(): void {
+    this.renderer.setStyle(this.elementRef.nativeElement, 'left', `0px`);
+    this.renderer.setStyle(this.elementRef.nativeElement, 'top', `0px`);
+    this.renderer.setStyle(this.elementRef.nativeElement, 'position', `absolute`);
+    this.renderer.setStyle(
+      this.elementRef.nativeElement,
+      'z-index',
+      DesignTokenHelper.zLayer('popover')
+    );
+  }
+
+  private updateHostElementPosition(): void {
+    this.config = {
+      placement: this.placement ?? 'bottom-start',
+      middleware: this.getMiddlewareConfig(),
+      strategy: 'absolute',
+    } as ComputePositionConfig;
+
+    computePosition(this.reference.nativeElement, this.elementRef.nativeElement, this.config).then(
+      ({ x, y }) => this.setPositionStylingOnHostElement(x, y)
+    );
+  }
+
+  private getMiddlewareConfig(): Array<Middleware | null | undefined | false> {
+    const middleware: Array<Middleware | null | undefined | false> = [];
+    middleware.push(offset(this.offset));
+
+    if (this.shift) {
+      middleware.push(shift());
+    }
+
+    if (this.autoPlacement) {
+      middleware.push(autoPlacement());
+    }
+
+    return middleware;
+  }
+
+  private setPositionStylingOnHostElement(xPosition: number, yPosition: number) {
+    this.renderer.setStyle(this.elementRef.nativeElement, 'left', `${xPosition}px`);
+    this.renderer.setStyle(this.elementRef.nativeElement, 'top', `${yPosition}px`);
+    this.renderer.setStyle(
+      this.elementRef.nativeElement,
+      'display',
+      this.isShown ? `block` : `none`
+    );
+  }
+
+  private setupEventHandling(): void {
+    if (!this.reference || !this.triggers) {
+      return;
+    }
+
+    if (Array.isArray(this.triggers)) {
+      (this.triggers as Array<TriggerEvent>).forEach((trigger: TriggerEvent) =>
+        this.attachTriggerEventToReferenceElement(trigger)
+      );
+    } else {
+      this.attachTriggerEventToReferenceElement(this.triggers);
+    }
+  }
```

Comment:
If we always have triggers as an array as mentioned above this could probably be 

```
this.triggers.forEach((trigger: TriggerEvent) => 
        this.attachTriggerEventToReferenceElement(trigger)
);

```